File: src/Application Specific/Classes/Entity Collections/Checklist/Task/TaskValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Tasks
    {
        // make sure that contact object is valid
        public class TaskValidator : BaseValidator
        {
            public void Validate(Task oTask)
            {

                // make sure that contact object is valid
                if (oTask.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oTask.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oTask.Description.Trim().Length == 0)
                {
                    AddCriticalMessage("Description Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Checklist/Task/TaskValidator.vb
Namespace Entity

Namespace Tasks

Public Class TaskValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oTask As Task)

        'make sure that contact object is valid
        If oTask.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oTask.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                If oTask.Description.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Description Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace
